process/block/preprocess/transactions.go

```diff
@@ -580,6 +580,27 @@ func (txs *transactions) CreateAndProcessMiniBlock(
 		}
 
 		snapshot := txs.accounts.JournalLen()
+		oldGasConsumedByMiniBlockInSenderShard := gasConsumedByMiniBlockInSenderShard
```

Comment:
can we add an integration test ? Creating a block which reaches the gasLimit and processing it should reach the same results.

Reply:
Added